Realm/RLMArray.mm

```diff
@@ -215,7 +215,7 @@ void RLMArrayValidateMatchingObjectType(__unsafe_unretained RLMArray *const arra
         @throw RLMException(@"Invalid nil value for array of '%@'.",
                             array->_objectClassName ?: RLMTypeToString(array->_type));
     }
-    if (array->_type != RLMPropertyTypeObject) {
+    if ((array->_type == RLMPropertyTypeAny) || (array->_type != RLMPropertyTypeObject)) {
```

Comment:
The first half of this conditional isn't doing anything. `array->_type == RLMPropertyTypeAny` can never be true without `array->_type != RLMPropertyTypeObject` also being true.